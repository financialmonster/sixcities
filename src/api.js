import axios from 'axios';

const api = axios.create({
	baseURL: `https://htmlacademy-react-3.appspot.com/six-cities`,
    timeout: 5000,
    withCredentials: true
});

const getOffers = () => api.get(`/hotels`);

const sendUserData = (userData) => api.post(`/login`, userData);

const getComments = (id) => api.get(`/comments/${id}`);

const sendUserReview = (review, id) => api.post(`/comments/${id}`, review);

const getFavorites = () => api.get(`/favorite`);

const setFavorite = (id, isFavorite) => api.post(`/favorite/${id}/${(isFavorite) ? 0 : 1}`);

export {getOffers, sendUserData, getComments, sendUserReview, getFavorites, setFavorite};
