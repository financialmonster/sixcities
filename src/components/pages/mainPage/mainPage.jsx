import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import CardList from '../../cardList/cardList.jsx';
import Header from '../../header/header.jsx';
import Navigation from '../../navigation/navigation.jsx';
import Filter from '../../filter/filter.jsx';
import {hasOffers} from '../../../selectors.js';
import Map from '../../map/map.jsx';

const MainPage = ({hasOffers}) => {
    const divClassName = (!hasOffers)
        ? `cities__places-container container cities__places-container--empty`
        : `cities__places-container container`;
    const sectionClassName = (!hasOffers)
        ? `cities__no-places`
        : `cities__places places`;

    return (
        <div className="page page--gray page--main">
            <Header />

            <main className="page__main page__main--index">
                <Navigation />

                <div className="cities">
                    <div className={divClassName}>
                        <section className={sectionClassName}>
                            {(!hasOffers) ? null : <Filter />}
                            
                            <CardList />
                        </section>
                        <Map />
                    </div>
                </div>
            </main> 
        </div>
    );
}

MainPage.propTypes = {
    hasOffers: PropTypes.bool.isRequired
}

const mapStateToProps = (state) => {
    return {
        hasOffers: hasOffers(state)
    };
}

export {MainPage};
export default connect(mapStateToProps, () => ({}))(MainPage);
