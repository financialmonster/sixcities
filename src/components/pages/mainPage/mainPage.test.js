import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';
import thunk from 'redux-thunk';

import api from '../../../api.js';
import {MainPage} from './mainPage.jsx';

describe(`Correctly renders the MainPage component `, () => {
    it(`when isLoading property is equal true and hasOffers property is equal false`, () => {
        const mockStore = configureStore([thunk.withExtraArgument(api)]);
        const store = mockStore({
            offers: {offers: [], isLoading: true, isError: false, currentSorting: `Popular`},
            card: {focusedCard: null},
            auth: {isLoggedIn: false, isAuthStatusLoading: false , isAuthStatusError: false}
        });

        const tree = renderer.create(
            <Provider store={store}>
                <Router>
                    <MainPage isLoading={true} hasOffers={false} />
                </Router>
            </Provider>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isLoading property is equal false and hasOffers property is equal true`, () => {
        const mockStore = configureStore([thunk.withExtraArgument(api)]);
        const store = mockStore({
            offers: {offers: [], isLoading: true, isError: false, currentSorting: `Popular`},
            card: {focusedCard: null, isLoading: false, isError: false},
            auth: {isLoggedIn: false, isAuthStatusLoading: false , isAuthStatusError: false}
        });

        const tree = renderer.create(
            <Provider store={store}>
                <Router>
                    <MainPage isLoading={false} hasOffers={true} />
                </Router>
            </Provider>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

     it(`when both hasOffers and isLoading properties are equal false`, () => {
        const mockStore = configureStore([thunk.withExtraArgument(api)]);
        const store = mockStore({
            offers: {offers: [], isLoading: false, isError: false, currentSorting: `Popular`},
            card: {focusedCard: null},
            auth: {isLoggedIn: false, isAuthStatusLoading: false , isAuthStatusError: false}
        });

        const tree = renderer.create(
            <Provider store={store}>
                <Router>
                    <MainPage isLoading={false} hasOffers={false} />
                </Router>
            </Provider>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
