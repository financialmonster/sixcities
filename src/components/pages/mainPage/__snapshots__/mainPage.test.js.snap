// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders the MainPage component  when both hasOffers and isLoading properties are equal false 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
                title="To the login page"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container cities__places-container--empty"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              No places to stay available
            </b>
            <p
              className="cities__status-description"
            >
              We could not find any property available at the moment in 
            </p>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <div
              id="map"
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`Correctly renders the MainPage component  when isLoading property is equal false and hasOffers property is equal true 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
                title="To the login page"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            0
             places to stay in 
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by 
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
              title="Open the menu"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
          </form>
          <div
            className="spinner"
          >
            <div
              className="spinnerContent"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </div>
        </section>
        <div
          className="spinner"
        >
          <div
            className="spinnerContent"
          >
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`Correctly renders the MainPage component  when isLoading property is equal true and hasOffers property is equal false 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
                title="To the login page"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <div
            className="spinner"
          >
            <div
              className="spinnerContent"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </div>
        </section>
        <div
          className="spinner"
        >
          <div
            className="spinnerContent"
          >
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
